SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -54,3 +54,43 @@ extension Array {
         }
     }
 }
+
+extension CollectionType {
+    public var ip_middleValues: [Generator.Element]? {
+        guard count > 0 else { return nil }
+        let needsAverageOfTwo = count.toIntMax().ip_isEven
+        let middle = startIndex.advancedBy(count / 2)
+        if needsAverageOfTwo {
+            let leftOfMiddle = startIndex.advancedBy((count / 2) - 1)
+            return [self[middle], self[leftOfMiddle]]
+        } else {
+            return [self[middle]]
+        }
+    }
+}
+
+extension SequenceType where Generator.Element: Equatable {
+    public func ip_mostCommonElements() -> [Generator.Element]? {
+        let sortedUniqueElements = self.ip_uniqueValues().sort {
+                self.ip_countOfElement($0) > self.ip_countOfElement($1)
+            }
+        guard let first = sortedUniqueElements.first else { return nil }
+        return sortedUniqueElements.lazy.filter {
+            self.ip_countOfElement(first) == self.ip_countOfElement($0)
+        }
+    }
+        
+    public func ip_uniqueValues() -> [Generator.Element] {
+        var buffer: [Generator.Element] = []
+        forEach { element in
+            if !buffer.contains(element) {
+                buffer.append(element)
+            }
+        }
+        return buffer
+    }
+    
+    public func ip_countOfElement(element: Generator.Element) -> Int {
```

Comment:
I think this should be `ip_countOf` to have better parity w/ swift apis, ie: `indexOf`